src/main/java/seedu/equipment/EquipmentManager.java

```diff
@@ -18,6 +18,14 @@ public void addEquipment(String itemName, String serialNumber, EquipmentType typ
         }
     }
 
+    public void addEquipment(Equipment equipment) throws DuplicateSerialNumber {
+        if (!equipmentList.containsKey(equipment.getSerialNumber())) {
+            equipmentList.putIfAbsent(equipment.getSerialNumber(), equipment);
+        } else if (equipmentList.containsKey(equipment.getSerialNumber())) {
+            throw new DuplicateSerialNumber();
+        }
```

Comment:
Maybe this part can be less verbose? E.g. `String serialNumber = equipment.getSerialNumber();`

Also, do we need the second `else if`? Since it's the other condition anyway - I'm not very sure on this w.r.t coding standards.

Reply:
Correct me if I'm wrong, but doesn't equipmentList.putIfAbsent() already check for whether the equipmentList contains the serial number? So technically we do not need the if condition on line 22?

Regarding the second else if, I think we can use a guard clause for better code quality.

Reply:
OK! Will change the else if into else. Also I changed into put instead of putIfAbsent since we checked the absence of it on the if else clause.